core/node/da_clients/src/avail/sdk.rs

```diff
@@ -450,13 +450,24 @@ impl GasRelayClient {
             .post(&submit_url)
             .body(Bytes::from(data))
             .header("Content-Type", "text/plain")
-            .header("Authorization", &self.api_key)
+            .header("Authorization", format!("Bearer {}", self.api_key))
             .send()
-            .await?;
+            .await
+            .map_err(|e| anyhow::anyhow!("Failed to submit data to the gas relay: {:?}", e))?;
 
-        let submit_response = submit_response
-            .json::<GasRelayAPISubmissionResponse>()
-            .await?;
+        let response_bytes = submit_response
+            .bytes()
+            .await
+            .context("Failed to read response body")?;
+
+        let submit_response =
+            match serde_json::from_slice::<GasRelayAPISubmissionResponse>(&response_bytes) {
+                Ok(response) => response,
+                Err(_) => {
+                    let response_body = String::from_utf8_lossy(&response_bytes).to_string();
```

Comment:
Nit: `to_string()` looks redundant; a `Cow` returned from `from_utf8_lossy()` has almost the same debug representation as the original string (for the identical one, the underlying`&str` can be borrowed via a deref or `as_ref()`).